### PR TITLE
Update docs website

### PR DESCRIPTION
**Pull request by pawelprazak**
- added Basics page
- improved linking between pages
- minor improvements to most of the pages
- missing.md update
- address #211

### CODE REVIEW

File: website/docs/basics.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+provides the implementation details. If not specified explicitly, the default provider is used.
+Providers can be configured using [provider configuration](https://www.pulumi.com/docs/concepts/resources/providers/#explicit-provider-configuration).
+
+#### Inputs and Outputs
+
+Inputs and Outputs are the primary [asynchronous data types in Pulumi](https://www.pulumi.com/docs/concepts/inputs-outputs/), 
+and they signify values that will be provided by the engine later, when the resource is created and its properties can be fetched.
+`Input[A]` type is an alias for `Output[A]` type used by [resource](#resources) arguments.
+
+Outputs are values of type `Output[A]` and behave very much like [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)).
```
Comment (pawelprazak) [inline website/docs/basics.md:118]: "like" monads or Besom Output is an actual monad?

File: website/docs/basics.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+```
+
+If you have multiple outputs and need to use them together you can use the standard 
+[`zip`](https://scala-lang.org/api/3.x/scala/collection/View.html#zip-1dd) 
+method to combine them into a single output:
+
+```scala
+val port: Output[Int] = pod.port
+val host: Output[String] = node.hostname
+val hello = host.zip(port).map((a, b) => s"https://$hostname:$port/")
```
Comment (pawelprazak) [inline website/docs/basics.md:145]: not 100% sure about the syntax

File: website/docs/basics.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+- [`pulumi config`](https://www.pulumi.com/docs/concepts/config/#setting-and-getting-configuration-values) command
+
+##### Accessing Configuration from Code
+
+Configuration values can be [accessed](https://www.pulumi.com/docs/concepts/config/#code) from [programs](#programs) 
+using the `Config` object, e.g.:
+
+```scala
+val a = config.get("aws:region")
+val b = Config("aws").map(_.get("region"))
```
Comment (pawelprazak) [inline website/docs/basics.md:179]: #204

File: website/docs/basics.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+##### Stack References
+
+[Stack References](https://www.pulumi.com/docs/concepts/stack/#stackreferences) allow you to use outputs from other 
+[stacks](#stacks) in your [program](#programs).
+
+To reference values from another stack, create an instance of the `StackReference` type using the fully qualified 
+name of the stack as an input, and then read exported stack outputs by their name.
+
+`StackReference` is not implemented yet in Besom, coming soon.
```
Comment (pawelprazak) [inline website/docs/basics.md:90]: #209

File: website/docs/basics.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+Resources represent the fundamental units that make up your infrastructure, such as a compute instance,
+a storage bucket, or a Kubernetes cluster.
+
+Resources are defined using a [**resource constructor**](constructors). Each resource in Pulumi has:
+- a [logical name and a physical name](https://www.pulumi.com/docs/concepts/resources/names/#resource-names)
+  The logical name establishes a notion of identity within Pulumi, and the physical name is used as identity by the provider
+- a [resource type](https://www.pulumi.com/docs/concepts/resources/names/#types), which identifies the provider and the kind of resource being created
+- [URN](https://www.pulumi.com/docs/concepts/resources/names/#types), which is an automatically constructed globally unique identifier for the resource.
+
+Each resource also can have:
```
Comment (KacperFKorban) [inline website/docs/basics.md:140]: typo: `can also have`

File: website/docs/basics.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+- a set of [arguments](https://www.pulumi.com/docs/concepts/resources/properties/) that define the behavior of the resulting infrastructure
+- and a set of [options](https://www.pulumi.com/docs/concepts/options/) that control how the resource is created and managed by the Pulumi engine.
+
+Every is managed by a [provider](#providers) which is a plugin that
```
Comment (KacperFKorban) [inline website/docs/basics.md:144]: typo: `resource`?

File: website/docs/basics.md (hunk 1 above)
Comment (KacperFKorban) [reply website/docs/basics.md:154]: Hmmm, `Output` has a bit of a complicated semantic. I think that it wouldn't be a crime to call them monads. The only benefit we get from calling it a monad is the analogy. I don't see anyone using generic functions over Monads for `Output`.
Maybe a useful analogy would be to say that it is besom's effect type?

File: website/docs/basics.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+val host: Output[String] = node.hostname
+val hello = host.zip(port).map { case (a, b) => s"https://$hostname:$port/" }
+```
+
+If you have a map of outputs and need to use them together **as a map** you can use
+`Output.traverse` method to combine them into a single output:
+
+```scala
+val m: Map[String, Output[String]] = Map(pod.name -> pod.port)
+val o: Output[Map[String, String]] = m.traverse // we use the extension method here
```
Comment (KacperFKorban) [inline website/docs/basics.md:198]: minor: Hmmm, I think that it's a very unconventional way of using `treverse`. A `List` example will be more illustrative.

File: website/docs/basics.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+Outputs are used to:
+- automatically captures dependencies between [resources](#resources)
+- provide a way to express transformations on its value before it's known
+- deffer the evaluation of its value until it's known
+- track the _secretness_ of its value
+
+Output transformations available in Besom:
+- [`map` and `flatMap`](apply_methods) methods take a callback that receives the plain value, and computes a new output
+- [lifting](lifting) directly read properties off an output value
+- [interpolation](interpolator) concatenate string outputs with other strings directly
```
Comment (KacperFKorban) [inline website/docs/basics.md:167]: nit: I would also just mention the methods from below here

File: website/docs/basics.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+val world = Output.secret("world")
+```
+
+If you have multiple outputs of the same type and need to use them together **as a list** you can use 
+`Output.sequence` method to combine them into a single output:
+
+```scala
+val port: Output[String] = pod.name
+val host: Output[String] = node.hostname
+val hello = List(host, port).sequence // we use the extension method here
```
Comment (KacperFKorban) [inline website/docs/basics.md:181]: nit: It would be super useful to add a type ascription here

File: website/docs/basics.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+with [Pulumi Service](https://www.pulumi.com/docs/intro/cloud-providers/pulumi-service/) being the default.
+
+State is used to:
+- track [resources](#resources) that are created by your [program](#programs)
+- record the relationship between resources
+- store metadata about your [project](#projects) and [stacks](#stacks)
+- and store [configuration](#configuration) and secret values
+- and store [stack outputs](#stack-outputs)
+
+:::tip
```
Comment (KacperFKorban) [inline website/docs/basics.md:265]: nit: Maybe downgrade this to `note`?

File: website/docs/basics.md [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+Pulumi is a registered trademark of [Pulumi Corporation](https://pulumi.com).
+
+### What is Besom?
+
+Besom is a **Pulumi SDK for Scala 3**. It allows you to use Scala to define your infrastructure
+in a type-safe and functional way.
+
+Besom **does NOT depend on Pulumi Java SDK**, it is a completely separate implementation.
+
+:::tip
```
[the remark this answers is not in the document]
Comment (pawelprazak) [reply website/docs/basics.md:23]:
```suggestion
:::caution
```

File: website/docs/basics.md [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+```
+
+#### Programs
+
+A Pulumi program, written in a general-purpose programming language, is a collection of [resources](#resources) 
+that are deployed to a [stack](#stacks).
+
+A minimal Besom program consists of:
+* `project.scala` - the program dependencies (here we use [Scala-CLI directives](https://scala-cli.virtuslab.org/docs/guides/using-directives/))
+    ```
```
[the remark this answers is not in the document]
Comment (pawelprazak) [reply website/docs/basics.md:71]: I've tried that, but it got rendered light gray for some reason and I thought it was a bit too light, but I'm happy to change this.

[the remark this answers is not in the document]
Comment (pawelprazak) [reply website/docs/basics.md:71]:
```suggestion
    ```scala
```

File: website/docs/basics.md (hunk 5 above)
Comment (pawelprazak) [reply website/docs/basics.md:140]:
```suggestion
Each resource can also have:
```

File: website/docs/basics.md (hunk 6 above)
Comment (pawelprazak) [reply website/docs/basics.md:144]:
```suggestion
Every [resource](#resources) is managed by a [provider](#providers) which is a plugin that
```

File: website/docs/basics.md (hunk 1 above)
Comment (pawelprazak) [reply website/docs/basics.md:154]: OK then, I'll leave as-is, this is the language Pulumi is using in their docs "like monad"

File: website/docs/basics.md (hunk 9 above)
Comment (pawelprazak) [reply website/docs/basics.md:181]:
```suggestion
val hello: Output[List[String]] = List(host, port).sequence // we use the extension method here
```

File: website/docs/basics.md (hunk 7 above)
Comment (pawelprazak) [reply website/docs/basics.md:198]: I'm sorry, I'm not sure hot to improve this line. I'll leave as-is in the interest of delivering this content to the users ASAP. Please let me know how can I make this line better.

File: website/docs/basics.md (hunk 10 above)
Comment (pawelprazak) [reply website/docs/basics.md:265]:
```suggestion
:::note
```

File: website/docs/basics.md (hunk 2 above)
Comment (KacperFKorban) [reply website/docs/basics.md]: it looks correct
maybe we should consider using mdoc to type-check the snippets?

File: website/docs/basics.md (hunk 7 above)
Comment (KacperFKorban) [reply website/docs/basics.md:209]: I would suggest sth like:
```scala
val names: List[String] = List("John", "Paul")
val outputNames: Output[List[String]] = names.traverse(name => Output(name))
```

File: website/docs/basics.md (hunk 2 above)
Comment (pawelprazak) [reply website/docs/basics.md]: Checking snippets sounds great! Yes, there was a typo.

This compiles:
```
  val port: Output[Int] = Output(8080)
  val host: Output[String] = Output("localhost")
  val hello = host.zip(port).map((hostname, port) => s"https://$hostname:$port/")
```

I've also improved the snippet a bit.

**Review by KacperFKorban — APPROVED** [review]

File: website/docs/basics.md (hunk 7 above)
Comment (pawelprazak) [reply website/docs/basics.md:209]: Actually, the map example is what users might be looking for (other SDKs tend to support them), but I didn't I can use traverse like that so this is also very useful, thank you

File: website/docs/basics.md (hunk 8 above)
Comment (pawelprazak) [reply website/docs/basics.md:167]: done :)

File: website/docs/basics.md (hunk 2 above)
Comment (pawelprazak) [reply website/docs/basics.md]: I've change the snipped to be more clear:
```
val port: Output[Int] = pod.port
val host: Output[String] = node.hostname
val hello: Output[(String, Int)] = host.zip(port)
val url: Output[String] = hello.map { case (a, b) => s"https://$hostname:$port/" }
```

**Pull request merged**